htlcswitch/switch.go

```diff
@@ -524,7 +528,14 @@ func (s *Switch) GetAttemptResult(attemptID uint64, paymentHash lntypes.Hash,
 			return
 		}
 		resultChan <- result
-	}()
+	})
+	// The switch shutting down is signaled by closing the channel.
+	if errors.Is(err, fn.ErrStopping) {
+		close(resultChan)
+	} else if err != nil {
+		return nil, fmt.Errorf("got an unexpected error from "+
+			"GoroutineManager: %w", err)
+	}
```

Comment:
related to my comment above: from what I can tell, the only error that `gm.Go(..)` will _ever_ return is `fn.ErrStopping` or `nil`. So to me, it returning this is not actually an error but more just a "state transition" we should be aware of. Which I think is a point towards handling this explicitly in the actual callback passed to `Go` via a quit channel as mentioned above. 
If we do want some idea of if the goroutine manager did its thing from outside the call-back (cause it could also be that the call-back never gets called), then I think a simple `bool` could to the trick since it basically just communicates "handled/not handled due to shutdown"